core/web/migrate_controller.go

```diff
@@ -477,6 +487,7 @@ func BuildTaskDAG(js models.JobSpec, tpe job.Type) (string, *pipeline.Pipeline,
 				pathString := path.String()
 
 				if path.IsArray() {
+					warnJSONPathDotSeparator = true
```

Comment:
Maybe this case isn't necessary...
@PiotrTrzpil If this hard coded path exists, would it always be used? (b/c it is the 'right' side of a merge later). In that case, we don't have to worry about an external dependency using `.` here IIUC.